bigsi/bloom/bit_matrix_reader.py

```diff
@@ -0,0 +1,66 @@
+import os
+import math
+from bitarray import bitarray
+
+ROWS_PER_SLICE = 80  # must be divisible by 8. somewhere around 80 seems optimal
```

Comment:
I don't understand why 80 is optimal, but I understand it is good to read several rows with a single disk read operation, and cache them into memory. I think however this number is variable, 80 rows can be few data if we have 1 sample, or lots of data if we have thousands of samples. Getting this correct number is annoying though (we might need to cache the disk block size per read?), so I think in the end that 80 is fine. Amazing that you cared about disk performance in this first implementation of these features.

Reply:
I did a profiling on different number of the dataset (1-2000 columns per input) and it looks like somewhere there gave me the best performance (about ~20% better than baseline). The `ROWS_PER_SLICE` mainly affects the creation of a new bitarray, reading the whole slice from file into the bitarray. So the number is really about the balance between reading frequency and reading chunk size, and taking into account the potential large number for the columns (hence the bits) like the 100k I built.

In terms of disk read, it was not much a concern for me as I believe the Python implementation of  file IO is buffered anyway.